tools/capable.py

```diff
@@ -149,12 +153,15 @@ def __getattr__(self, name):
     bpf_text = bpf_text.replace('FILTER1',
         'if (pid != %s) { return 0; }' % args.pid)
 if not args.verbose:
-    bpf_text = bpf_text.replace('FILTER2', 'if (audit == 0) { return 0; }')
+    bpf_text = bpf_text.replace('FILTER2', 'if ((cap_opt & 0b1) == 0) { return 0; }')
+if args.no_insetid:
+    bpf_text = bpf_text.replace('FILTER_INSETID', 'if ((cap_opt & 0b10) != 0) { return 0; }')
```

Comment:
In old kernel, we have
```
#define SECURITY_CAP_NOAUDIT 0
#define SECURITY_CAP_AUDIT 1
```
In the new kernel, we have
```
/* Default (no) options for the capable function */
#define CAP_OPT_NONE 0x0        
/* If capable should audit the security request */
#define CAP_OPT_NOAUDIT BIT(1) 
/* If capable is being called by a setid function */
#define CAP_OPT_INSETID BIT(2)
```

Looks like the above `cap_opt & ...` may not be correct, right? Testing for audit/noaudit is different for old and new kernels.

Can we include `linux/security.h` and using macros to differentiate old and new kernels, this way, we do not need option `no_insetid`?

In print out, if INSETID is not supported, we just print out N/A. In bpf program. The above macro should indicate whether INSETID is supported or not.

Reply:
Thanks for the review!
Yes, I didn't expect the logic to change that much and should have checked again.
I've fixed it and now it also makes sense to remove the `-n` option.